File: solutions/go/0712-minimum-ascii-delete-sum-for-two-strings.go
func minimumDeleteSum(s1 string, s2 string) int {
    prev_row := make([]int, len(s2) + 1)
    for j := 1; j <= len(s2); j++ {
        prev_row[j] = prev_row[j - 1] + int(s2[j - 1])
    }

    for i := 1; i <= len(s1); i++ {
        curr_row := []int{prev_row[0] + int(s1[i - 1])}
        for j := 1; j <= len(s2); j++ {
            if s1[i - 1] == s2[j - 1] {
                curr_row = append(curr_row, prev_row[j - 1])
            } else {
                curr_row = append(curr_row, min(prev_row[j] + int(s1[i - 1]), curr_row[j - 1] + int(s2[j - 1])))
            }
        }
        prev_row = curr_row
    }

    return prev_row[len(prev_row) - 1]
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: solutions/swift/0712-minimum-ascii-delete-sum-for-two-strings.swift
class Solution {
    func minimumDeleteSum(_ s1: String, _ s2: String) -> Int {
        var prevRow = [Int](repeating: 0, count: s2.count + 1)
        let s1 = Array(s1.unicodeScalars)
        let s2 = Array(s2.unicodeScalars)
        
        for j in 1...s2.count {
            prevRow[j] = prevRow[j - 1] + Int(s2[j - 1].value)
        }

        for i in 1...s1.count {
            var currRow = [prevRow[0] + Int(s1[i - 1].value)]
            for j in 1...s2.count {
                if s1[i - 1] == s2[j - 1] {
                    currRow.append(prevRow[j - 1])
                } else {
                    currRow.append(min(prevRow[j] + Int(s1[i - 1].value), currRow[j - 1] + Int(s2[j - 1].value)))
                }
            }
            prevRow = currRow
        }
        return prevRow.last!
    }
}
